syft/execution/role.py

```diff
@@ -111,10 +132,16 @@ def execute(self, args_):
     def _instantiate_inputs(self, args_):
         """ Takes input arguments for this role and generate placeholders.
         """
-        input_placeholders = tuple(
+
+        def traversal_function(obj):
+            placeholder = input_placeholders.pop(0)
+            placeholder.instantiate(obj)
```

Comment:
I'm not sure I have a better solution right now but this doesn't seem very clean

Reply:
It was the cleanest way to use a local reference, I could use `functools.partial` to be more explicit about it, but it won't be any fancier I think.